filebeat/crawler/crawler.go

```diff
@@ -82,4 +106,10 @@ func (c *Crawler) Stop() {
 	}
```

Comment:
can we change the go routine to:

```
go func(p Prospector){
    defer c.wg.Done()
    p.Stop
}(prospector)
```

The prospector shouldn't need to know someone is waiting for stop having finished. Simple form of abstraction leak.